github/resource_github_actions_secret.go

```diff
@@ -0,0 +1,157 @@
+package github
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"github.com/google/go-github/v29/github"
+	"github.com/hashicorp/terraform/helper/schema"
+	"golang.org/x/crypto/nacl/box"
+	"log"
+)
+
+func resourceGithubActionsSecret() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubActionsSecretCreateOrUpdate,
+		Read:   resourceGithubActionsSecretRead,
+		Update: resourceGithubActionsSecretCreateOrUpdate,
+		Delete: resourceGithubActionsSecretDelete,
+
+		Schema: map[string]*schema.Schema{
+			"owner": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"repository": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"secret_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"plaintext_value": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func resourceGithubActionsSecretCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	err := checkOrganization(meta)
+	if err != nil {
+		return err
+	}
+
+	client := meta.(*Organization).client
+	ctx := context.Background()
+
+	repo := d.Get("repository").(string)
+	owner := d.Get("owner").(string)
+	secretName := d.Get("secret_name").(string)
+	plaintextValue := d.Get("plaintext_value").(string)
+
+	keyId, publicKey, err := getPublicKeyDetails(owner, repo, meta)
+	if err != nil {
+		return err
+	}
+
+	encryptedText, err := encryptPlaintext(plaintextValue, publicKey)
+	if err != nil {
+		return err
+	}
+
+	// Create an EncryptedSecret and encrypt the plaintext value into it
+	eSecret := &github.EncryptedSecret{
+		Name:           secretName,
+		KeyID:          keyId,
+		EncryptedValue: base64.StdEncoding.EncodeToString(encryptedText),
+	}
+
+	_, err = client.Actions.CreateOrUpdateSecret(ctx, owner, repo, eSecret)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s:%s", owner, repo, secretName))
+
+	return resourceGithubActionsSecretRead(d, meta)
+}
+
+func resourceGithubActionsSecretRead(d *schema.ResourceData, meta interface{}) error {
+	err := checkOrganization(meta)
+	if err != nil {
+		return err
+	}
+
+	client := meta.(*Organization).client
+	ctx := context.Background()
+
+	repo := d.Get("repository").(string)
+	owner := d.Get("owner").(string)
+	secretName := d.Get("secret_name").(string)
+
+	secret, _, err := client.Actions.GetSecret(ctx, owner, repo, secretName)
+	if err != nil || secret.Name != secretName {
+		d.SetId("")
+		return err
+	}
+
+	return nil
+}
+
+func resourceGithubActionsSecretDelete(d *schema.ResourceData, meta interface{}) error {
+	err := checkOrganization(meta)
+	if err != nil {
+		return err
+	}
+
+	client := meta.(*Organization).client
+	ctx := context.WithValue(context.Background(), ctxId, d.Id())
+
+	orgName := d.Get("owner").(string)
+	repoName := d.Get("repository").(string)
```

Comment:
Perhaps it's better to obtain `repoName` and `secretName` by extracting it from `d.GetId()` using [`parseTwoPartID`](https://github.com/terraform-providers/terraform-provider-github/blob/b838aed6391ac450740f023e4a714d6cd449aad3/github/util.go#L49-L56) utility func?

```go
repoName, secretName, err := parseTwoPartID(d.Id(), "repository", "secret_name")
if err != nil {
    return err
}
```